### ISSUE TITLE
Error when using mapTo on a property of type List

### ISSUE THREAD

**Issue by GregoryEsberci** [realm/realm-js#6268]
### How frequently does the bug occur?

Always

### Description

When mapTo is used on a property of type List, an error occurs when trying to access it.

### Stacktrace & log output

```shell
ERROR  [Error: Property 'test_list' does not exist on 'Task' objects] 
    at TaskList (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.ExampleRnTodoList:162361:21)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.ExampleRnTodoList:59191:43)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.ExampleRnTodoList:59191:43)
    at TaskScreen (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.ExampleRnTodoList:161948:22)
    at anonymous (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.ExampleRnTodoList:123651:27)
    at RCTSafeAreaView
    at AppNonSync
    at App
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.ExampleRnTodoList:59191:43)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.ExampleRnTodoList:59191:43)
    at AppContainer (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&inlineSourceMap=false&modulesOnly=false&runModule=true&app=org.reactjs.native.example.ExampleRnTodoList:59035:36)
    at ExampleRnTodoList(RootComponent) (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&inlineSour
```

### Can you reproduce the bug?

Always

### Reproduction Steps

```js
class Task extends Realm.Object {
  _id: BSON.ObjectId = new BSON.ObjectId();
  description!: string;
  @mapTo('complete')
  isComplete: boolean = false;
  createdAt: Date = new Date();
  userId!: string;

  @mapTo('test_list')
  testList!: Realm.List<string>;

  testList2!: Realm.List<string>;

  static primaryKey = '_id';

  constructor(realm, description, userId) {
    super(realm, {
      description,
      userId: userId || '_SYNC_DISABLED_',
    });
  }
}
```

```js
// OK
console.log('isComplete', task.isComplete);

try {
  // Error
  console.log('testList', task.testList);
} catch (error) {
  console.error(error);
}

try {
  // OK
  console.log('testList2', task.testList2);
} catch (error) {
  console.error(error);
}
```

### Version

12.3.0

### What services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

ios 16.4

### Build environment

React Native: 0.72.6

**Comment by kneth** [comment]
@GregoryEsberci Thank you for reporting.

In https://github.com/realm/realm-js/pull/6269 we are trying to reproduce it but we still need a couple to test cases to add - stay tuned.

**Comment by kneth** [comment]
@GregoryEsberci I believe we have a fix in #6269

**Comment by GregoryEsberci** [comment]
Thank you very much for the quick solution to the problem!
